cranelift-codegen/meta-python/isa/x86/settings.py

```diff
@@ -35,6 +35,13 @@
 use_bmi1 = And(has_bmi1)
 use_lzcnt = And(has_lzcnt)
 
+is_pic = And(shared.is_pic)
+not_is_pic = Not(shared.is_pic)
+all_ones_funcaddrs_and_not_is_pic = And(shared.allones_funcaddrs,
+                                        Not(shared.is_pic))
+not_all_ones_funcaddrs_and_not_is_pic = And(Not(shared.allones_funcaddrs),
+                                            Not(shared.is_pic))
```

Comment:
Can you comment on why `is_pic` and other target-independent settings are duplicated here in the x86-specific settings.py, and similarly below in x86/mod.rs?

Reply:
Yes, the Python code doesn't make it quite obvious: `shared.is_pic` is a BooleanSetting, while `is_pic` is a predicate. The python code uses the `And` function with one argument to transform a setting into a predicate; the Rust code is more verbose and explicit, e.g. `settings.add_predicate("is_pic", predicate!(shared.get_bool("is_pic")))` does the same thing in Rust code.

Then, it is locally duplicated in the target specific settings definition, because predicates belong to a `SettingGroup`, which are tied to the target ISA. Before this PR, the same predicates would end up in the setting group, but were anonymous (that is, instruction predicates would use the settings and the predicate would be created on the fly). This change makes them explicit (by naming them) so we just remove the concept of anonymous predicates (and don't have to port it, when porting encodings).

Do you want me to add a comment around here to explain this, or in the Rust crate?

Reply:
A comment in the Rust crate would be good.

Reply:
Done!